git.go

```diff
@@ -82,31 +88,36 @@ func (scm *GitScm) calculateCommits() (*object.Commit, *object.Commit, error) {
 	return headCommit, targetCommit, nil
 }
 
-// checkGitProvider identies the head sha and target branch from the environment variables that are
+// checkGiContext identifies the head sha and target branch from the environment variables that are
 // populated from a Git provider, such as Github or Gitlab. If no proprietary env vars are set, then it will
 // look up this tool-specific variable for the target branch.
-func checkGitProvider() (string, string, string, bool) {
+func checkGiContext() *ScmContext {
+	// in local branches, we are not in pull/merge requests
+	localContext := FromLocal()
```

Comment:
We prioritise the existence of the tool's own variables to be read first. This is allowing, i.e., running tests for each SCM context using Github actions